pkg/network/ebpf/c/http.h

```diff
@@ -215,7 +174,10 @@ static __always_inline int http_process(http_transaction_t *http_stack, skb_info
         return 0;
     }
 
-    http_transaction_t *to_flush = http_should_flush_previous_state(http, packet_type);
+    if (http_should_flush_previous_state(http, packet_type)) {
+        http_enqueue(http);
```

Comment:
no, there will be at most one call to `http_enqueue` per `http_process` invocation.
prior to this change inlining `http_enqueue` was too costly because the function was huge due to the unrolled loop that is now removed, so we had to structure the code in a weird way to work around that. now that this is no longer a concern I'm just reorganizing the code in a way that hopefully is easier to follow

Reply:
Yes, I see the comment now in the old code, so we're fairly certain `http_should_flush_previous_state` will return `NULL` for closed connections every time?

Reply:
yup!

Reply:
:+1: